### PR TITLE
Implement OptimizeResult.__dir__

### PR DESCRIPTION
**Pull request by anntzer**
This lets tab-completion of OptimizeResult objects suggest the actual fields set, rather than the fairly irrelevant methods of the underlying dict object.

In fact before this gets merged I'd like to get people's opinion on whether to include dict methods at all in the result, given that the goal is mostly interactive use; cf the Python docs regarding `dir`:

> Because dir() is supplied primarily as a convenience for use at an interactive prompt, it tries to supply an interesting set of names more than it tries to supply a rigorously or consistently defined set of names, and its detailed behavior may change across releases. For example, metaclass attributes are not in the result list when the argument is a class.

### CODE REVIEW

**Comment by larsmans** [comment]
Trivial as this might seem, I would like to see a simple test for it.

> I'd like to get people's opinion on whether to include dict methods at all in the result

Omitting them is fine with me.

**Comment by anntzer** [comment]
Here we go.

**Comment by larsmans** [comment]
LGTM.

File: scipy/optimize/optimize.py [hunk 1]
```diff
@@ -126,6 +126,9 @@ def __repr__(self):
         else:
             return self.__class__.__name__ + "()"
 
+    def __dir__(self):
+        return self.keys()
+
```
Comment (dlax) [inline scipy/optimize/optimize.py]: Looks like you need to return a sequence (not an iterable). https://docs.python.org/3/reference/datamodel.html#object.__dir__

**Comment by anntzer** [comment]
fixed.

**Pull request merged**